deluge/tests/test_torrentmanager.py

```diff
@@ -78,7 +78,7 @@ async def test_prefetch_metadata(self):
 
         magnet = 'magnet:?xt=urn:btih:ab570cdd5a17ea1b61e970bb72047de141bce173'
         d = self.tm.prefetch_metadata(magnet, 30)
-        self.tm.on_alert_metadata_received(mock_alert)
+        reactor.callLater(0, self.tm.on_alert_metadata_received, mock_alert)
```

Comment:
Here's something I hadn't considered with the maybe_coroutine stuff. I don't think it's necessarily bad, since it really only affects tests rather than real usage. Since the prefetech_metadata call is actually async now (deferred returning functions are a bit different, in that they run synchronously until a deferred gets returned,) it hasn't run at all at this point, and just firing the metadata received alert happened before the torrent even got added to the session. Scheduling it to run means it happens after the initial prefetch_metadata call is started, and the test works properly again.

Reply:
Could we add  this detail to the commit message and perhaps even a short comment summary above this line